src/hotspot/share/runtime/mutex.cpp

```diff
@@ -151,6 +154,25 @@ bool Mutex::try_lock() {
   return false;
 }
 
+bool Mutex::try_lock() {
+  Thread * const self = Thread::current();
+  if (_owner == self) {
+    return false;
+  }
```

Comment:
This is being fixed by https://github.com/openjdk/jdk/pull/1247

Reply:
I'll wait until you push and then I'll merge. I still have to check the owner here though, because if the thread already owns the lock, calling check_rank() from try_lock will do the wrong check (it will assume it's being called from wait()). An alternative would be to add a boolean parameter to check_rank(), set only from try_lock() so I can differentiate the two calls.

Reply:
How about just:
```
bool Mutex::try_lock() {
  Thread * const self = Thread::current();
  DEBUG_ONLY(if (!owned_by_self()) check_rank();)
  return try_lock_inner(self);
}
```
keeping try_lock_without_rank_check() as is. Then I can merge try_lock_inner() with your patch.

Reply:
Ok, makes sense. I updated the code with the suggested change.

Thanks!